exts/jphp-zend-ext/src/main/java/org/develnext/jphp/zend/ext/standard/ArrayFunctions.java

```diff
@@ -586,6 +587,23 @@ public static Memory array_keys(Environment env, TraceInfo trace, Memory input,
         return result.toConstant();
     }
 
+    public static Memory array_key_first(Environment env, TraceInfo trace, Memory input) {
+        return array_key_impl(input, true);
+    }
+
+    public static Memory array_key_last(Environment env, TraceInfo trace, Memory input) {
+        return array_key_impl(input, false);
+    }
+
+    private static Memory array_key_impl(Memory input, boolean first) {
+        if (!input.isArray() || empty(input)) {
+            return Memory.NULL;
+        }
+
+        ForeachIterator it = input.toValue(ArrayMemory.class).foreachIterator(false, false);
+        return (first ? it.next() : it.end()) ? it.getMemoryKey() : Memory.NULL;
```

Comment:
`it.end()` will iterate all items in array, see ItemsUtils class and its `last` and `lastKey` implementation.

Reply:
https://github.com/jphp-group/jphp/blob/d9b269526affe5d0dd763b53fcc7ba2bd07f6738/jphp-runtime/src/php/runtime/ext/core/classes/lib/ItemsUtils.java#L372

Reply:
@dim-s 

After changes made recently `it.getMemoryKey()` does not work properly.

```php
$array = array( "1" => "one", "2" => "two", "3" => "three", "4" => "four", "5" => "five" );
var_dump(array_key_first($array));
```
Expected: `int(1)`.
Actual: `string(1) "1"`.

Can you take a look?

Reply:
@nstdio I've fixed: https://github.com/jphp-group/jphp/commit/1fc9bed61e984180a9a76511bde95bdf3360d243